openhands_cli/conversations/lister.py

```diff
@@ -0,0 +1,202 @@
+        """Find the first user prompt in the conversation events.
+
+        Args:
+            event_files: List of event file paths sorted by name.
+
+        Returns:
+            First user prompt text or None if not found.
+        """
+        for event_file in event_files:
+            event_data = self._load_event_data(event_file)
+            if event_data is None:
+                continue
+
+            message_event = self._to_message_event(event_data)
+            if message_event is None or message_event.source != "user":
+                continue
+
+            text = self._extract_text_from_message_event(message_event)
+            if text:
+                return text
+
+        return None
+
+    def _load_event_data(self, event_file: Path) -> dict[str, Any] | None:
+        """Safely load JSON event data from a file."""
+        try:
+            with open(event_file, encoding="utf-8") as f:
+                return json.load(f)
+        except (OSError, json.JSONDecodeError, TypeError):
+            return None
+
+    def _to_message_event(self, event_data: dict[str, Any]) -> MessageEvent | None:
+        """Convert raw event data to a MessageEvent, handling old formats."""
+        # Try new format first
+        try:
+            return MessageEvent(**event_data)
+        except Exception:
+            pass
+
+        # Try transformed old format
+        try:
+            transformed_data = self._transform_old_message_format(event_data)
+            return MessageEvent(**transformed_data)
+        except Exception:
+            return None
+
+    def _extract_text_from_message_event(
+        self, message_event: MessageEvent
+    ) -> str | None:
+        """Extract the first non-empty text content from a MessageEvent."""
+        llm_message = getattr(message_event, "llm_message", None)
+        if llm_message is None:
+            return None
+
+        for content_item in getattr(llm_message, "content", []):
+            if isinstance(content_item, TextContent):
+                text = content_item.text.strip()
+                if text:
+                    return text
+        return None
+
+    def _transform_old_message_format(self, event_data: dict) -> dict:
+        """Transform old message format to new MessageEvent format.
+
+        Args:
+            event_data: Event data in old format.
+
+        Returns:
+            Event data in new format.
+        """
+        if (
+            event_data.get("kind") == "MessageEvent"
+            and "message" in event_data
+            and "llm_message" not in event_data
+        ):
+            transformed = event_data.copy()
+            # Move message to llm_message and add role
+            transformed["llm_message"] = event_data["message"].copy()
+            transformed["llm_message"]["role"] = event_data["source"]
+            del transformed["message"]
+            return transformed
+
+        return event_data
```

Comment:
I don't think we need to handle this since the Event structure doesn't change since we did the 1.0.0 release?